### ISSUE TITLE
Have Tidy handle "\r\n" line endings

### ISSUE THREAD

**Issue by sts10** [sts10/tidy#45]
Some word lists may have "/r/n" line endings, like [this list](https://www.eff.org/files/2018/08/29/gameofthrones_8k-2018.txt). It's be neat if Tidy could fix these line endings. 

I would have thought that Rust's File or BufReader objects would have an easy option for this, but I haven't found one.

**Comment by eaon** [comment]
Yep, `\r\n` is a Windows thing.

As to how to handle that nicely, anything that implements `std::io::BufRead` will have [`lines()`](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.lines) which is agnostic about line endings :slightly_smiling_face:

**Comment by sts10** [comment]
Ah, re-visiting this, I'm wondering if I described the issue correctly. 

For starters, I do use `lines()` [here](https://github.com/sts10/tidy/blob/691ed93166556e78c3a6d24a9ce6a1cda167e209/src/file_readers.rs#L24-L36), which makes me wonder if this example I found from EFF is a rare fluke...

What I'm wondering now is whether that particular EFF word list is somehow even more strange than a normal Windows text file. On Linux, running `curl https://www.eff.org/files/2018/08/29/gameofthrones_8k-2018.txt | wc -l` gives me `1`. I'm guessing that is not how incompatible ever Windows-created text file is on Linux/for `wc`?

If we did want Tidy to be able to deal with a one-line file like this, it would have to correctly detect the line endings itself I think? Obviously I'd rather not write my own version of the `lines()`... 

What we probably need is a solid test file that represents what a word list created on Windows is like. I'm guessing, from your comment and general robustness of Rust, that `lines() will indeed handle such a file just fine.

**Comment by eaon** [comment]
Yeah, that's an _odd_ file, and I would argue the file is "broken". It looks as if there's only 1 `\r\n` in there, the rest is exclusively `\r`, which makes no sense, as the use case for this list on an [Apple II or C64](https://en.wikipedia.org/wiki/Newline#Representation) is the kind of niche that it would only apply to a couple people on the planet. I suspect whoever last edited this list is maybe also a retro-computer enthusiast and forgot to change the settings for line endings - or simply accidentally clicked on the wrong option.

Either way, I would argue tidy isn't responsible for the correctness of the formatting of a file. That's the sort of thing that's easiest to fix with `tr`:

```console
$ curl -s https://www.eff.org/files/2018/08/29/gameofthrones_8k-2018.txt | tr '\r' '\n' | wc -l
8013
```

I think it'd make the most sense to explicitly call out that tidy supports `\n` and `\r\n` line endings, and that it's the users responsibility to ensure that a correct format is supplied.
